app/jobs/regular/update_username.rb

```diff
@@ -5,16 +5,27 @@ class UpdateUsername < Jobs::Base
 
     def execute(args)
       @user_id = args[:user_id]
-      @old_username = args[:old_username]
-      @new_username = args[:new_username]
+      @old_username = args[:old_username].unicode_normalize
+      @new_username = args[:new_username].unicode_normalize
       @avatar_img = PrettyText.avatar_img(args[:avatar_template], "tiny")
 
-      @raw_mention_regex = /(?:(?<![\w`_])|(?<=_))@#{@old_username}(?:(?![\w\-\.])|(?=[\-\.](?:\s|$)))/i
+      @raw_mention_regex = /
+        (?:
+          (?<![\p{Alnum}\p{M}`])     # make sure there is no preceding letter, number or backtick
+        )
+        @#{@old_username}
+        (?:
+          (?![\p{Alnum}\p{M}_\-.`])  # make sure there is no trailing letter, number, underscore, dash, dot or backtick
+          |                          # or
+          (?=[-_.](?:\s|$))          # or there is an underscore, dash or dot followed by a whitespace or end of line
+        )
+      /ix
```

Comment:
I converted this regex into free-spacing mode and added comments, because mentally parsing a complex expression isn't fun. 😃 And now it supports Unicode.

Reply:
It looks really nice!